stream-chat-android-client/src/main/java/io/getstream/chat/android/client/notifications/PushNotificationRenderer.kt

```diff
@@ -0,0 +1,244 @@
+package io.getstream.chat.android.client.notifications
+
+import android.app.Notification
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.media.RingtoneManager
+import androidx.core.app.NotificationCompat
+import androidx.core.app.RemoteInput
+import io.getstream.chat.android.client.R
+import io.getstream.chat.android.client.extensions.sendImplicitBroadcast
+import io.getstream.chat.android.client.receivers.NotificationMessageReceiver
+
+public open class PushNotificationRenderer : BroadcastReceiver() {
+
+    override fun onReceive(context: Context, intent: Intent) {
+        val notificationId = intent.getIntExtra(EXTRA_NOTIFICATION_ID, 0)
+        val channelName: String = intent.getStringExtra(EXTRA_CHANNEL_NAME)
+        val messageText = intent.getStringExtra(EXTRA_MESSAGE_TEXT)
+        val messageId = intent.getStringExtra(EXTRA_MESSAGE_ID)
+        val channelType = intent.getStringExtra(EXTRA_CHANNEL_TYPE)
+        val channelId = intent.getStringExtra(EXTRA_CHANNEL_ID)
```

Comment:
We can't do any background work in `BroadcastReceiver` and this class is intended just to show push notifications. `Channel` and `Message` are not serializable, therefore passing required data in `intent`. Customers might need more information here and request additional extra data fields.